pennylane/templates/embeddings.py

```diff
@@ -136,15 +140,22 @@ def AmplitudeEmbedding(features, wires):
     Args:
         features (array): Input array of shape ``(2**n,)``
         wires (Sequence[int]): sequence of qubit indices that the template acts on
+        pad (Boolean): controls the activation of the padding option, defaults to ``False``
     """
 
     if not isinstance(wires, Iterable):
         raise ValueError("Wires needs to be a list of wires that the embedding uses; got {}.".format(wires))
 
-    if 2**len(wires) != len(features):
-        raise ValueError("AmplitudeEmbedding requires a feature vector of size 2**len(wires), which is {}; "
+    if pad and 2**len(wires) != len(features):
+        features = np.pad(features, (0, 2**len(wires)-len(features)), 'constant')
+
+    if  not pad and 2**len(wires) != len(features):
+        raise ValueError("AmplitudeEmbedding with no padding requires a feature vector of size 2**len(wires), which is {}; "
                          "got {}.".format(2 ** len(wires), len(features)))
 
+    if np.linalg.norm(features, 2) != 1:
+        raise ValueError("AmplitudeEmbedding requires a normalized feature vector.")
```

Comment:
```suggestion
        raise ValueError("AmplitudeEmbedding requires a feature vector normalized to norm 1.")
```
Or something like it. We should express what kind of nomalization is required.